plugin/src/main/java/net/dzikoysk/funnyguilds/feature/placeholders/GuildPlaceholders.java

```diff
@@ -0,0 +1,168 @@
+                                .replace("{POINTS}", String.valueOf(guild.getRank().getAveragePoints())),
+                        () -> NumberRange.inRangeToString(0, config.pointsFormat)
+                                .replace("{POINTS}", "0"))
+                .property("kills", (guild, rank) -> rank.getKills(), () -> 0)
+                .property("avg-kills", (guild, rank) -> rank.getAverageKills(), () -> 0)
+                .property("deaths", (guild, rank) -> rank.getDeaths(), () -> 0)
+                .property("avg-deaths", (guild, rank) -> rank.getAverageDeaths(), () -> 0)
+                .property("assists", (guild, rank) -> rank.getAssists(), () -> 0)
+                .property("avg-assists", (guild, rank) -> rank.getAverageAssists(), () -> 0)
+                .property("logouts", (guild, rank) -> rank.getLogouts(), () -> 0)
+                .property("avg-logouts", (guild, rank) -> rank.getAverageLogouts(), () -> 0)
+                .property("kdr", (guild, rank) -> String.format(Locale.US, "%.2f", rank.getKDR()), () -> 0.0)
+                .property("avg-kdr", (guild, rank) -> String.format(Locale.US, "%.2f", rank.getAverageKDR()), () -> 0.0);
+
+        GUILD_ALLIES_ENEMIES_ALL = new GuildPlaceholders()
+                .property("allies",
+                        guild -> joinOrDefault.apply(Entity.names(guild.getAllies()), messages.alliesNoValue),
+                        () -> messages.alliesNoValue)
+                .property("allies-tags",
+                        guild -> joinOrDefault.apply(GuildUtils.getTags(guild.getAllies()), messages.alliesNoValue),
+                        () -> messages.alliesNoValue)
+                .property("enemies",
+                        guild -> joinOrDefault.apply(Entity.names(guild.getEnemies()), messages.enemiesNoValue),
+                        () -> messages.enemiesNoValue)
+                .property("enemies-tags",
+                        guild -> joinOrDefault.apply(GuildUtils.getTags(guild.getEnemies()), messages.enemiesNoValue),
+                        () -> messages.enemiesNoValue);
+
+        GUILD_MEMBERS_COLOR_CONTEXT = new Placeholders<Pair<String, Guild>>()
+                .property("members", pair -> {
+                    String text = Joiner.on(", ").join(UserUtils.getOnlineNames(pair.getSecond().getMembers())).toString();
+
+                    return !text.contains("<online>")
+                            ? text
+                            : ONLINE.toFormatter(pair.getFirst()).format(text);
+                });
+    }
+
+    public GuildPlaceholders property(String name, MonoResolver<Guild> resolver, SimpleResolver fallbackResolver) {
+        GuildPlaceholders copy = new GuildPlaceholders();
+        copy.placeholders.putAll(this.placeholders);
+        copy.placeholders.put("{" + name.toUpperCase() + "}", new FallbackPlaceholder<>(resolver, fallbackResolver));
```

Comment:
Może to by dać do jakiejś metody statycznej w Placeholders - wtedy dodawanie i usuwanie nawiasów odbywałoby się jedynie w dwóch miejscach